File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/BuilderBonusContaExecutivaService.java
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy;

import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta;

import java.math.BigDecimal;
import java.util.Objects;

public class BuilderBonusContaExecutivaService implements IFactoryObterPercentualBonusService {

    private static final double VALOR_FIXO_DEFAUL_PERCENTUAL_CINCO_PORCENTO = 5.00;

    @Override
    public boolean isAppliable(Conta conta) {
        return (Objects.nonNull(conta) && conta.isTipoContaIsExecutiva());
    }

    @Override
    public BigDecimal obterValorPercentualBonusAoDeposito(Conta conta) {
        return (isAppliable(conta)) ? BigDecimal.valueOf(VALOR_FIXO_DEFAUL_PERCENTUAL_CINCO_PORCENTO) : BigDecimal.ZERO;
    }

    @Override
    public PercentualBonus getPercentualBonus(Conta conta) {
        return (isAppliable(conta)) ? PercentualBonus.BONUS_CONTA_EXECUTIVA : PercentualBonus.BONUS_CONTA_NAO_ESPECIFICADA;
    }
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/service/strategy/BuilderBonusContaExecutivaService.kt
package br.com.blu.bs2.sintetizador.contas.csv.service.strategy

import br.com.blu.bs2.sintetizador.contas.csv.model.dominio.Conta
import br.com.blu.bs2.sintetizador.contas.csv.service.strategy.PercentualBonus
import java.math.BigDecimal
import java.util.*

class BuilderBonusContaExecutivaService : IFactoryObterPercentualBonusService {
    override fun isAppliable(conta: Conta?): Boolean {
        return Objects.nonNull(conta) && conta!!.isTipoContaIsExecutiva
    }

    override fun obterValorPercentualBonusAoDeposito(conta: Conta?): BigDecimal? {
        return if (isAppliable(conta)) BigDecimal.valueOf(VALOR_FIXO_DEFAUL_PERCENTUAL_CINCO_PORCENTO) else BigDecimal.ZERO
    }

    override fun getPercentualBonus(conta: Conta?): PercentualBonus? {
        return if (isAppliable(conta)) PercentualBonus.BONUS_CONTA_EXECUTIVA else PercentualBonus.BONUS_CONTA_NAO_ESPECIFICADA
    }

    companion object {
        private const val VALOR_FIXO_DEFAUL_PERCENTUAL_CINCO_PORCENTO = 5.00
    }
}
